WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -11,21 +11,36 @@
 
 namespace WalletWasabi.Services
 {
-	public class SystemAwakeChecker : PeriodicRunner
+	public class SystemAwakeChecker
 	{
 		private const string Reason = "CoinJoin is in progress.";
 		private static readonly TimeSpan Timeout = TimeSpan.FromMinutes(5);
 
+		private bool _canShutdown = true;
+		private object _shutDownLock = new();
```

Comment:
locks are supposed to be `readonly`.

Reply:
The variable name could probably be something like `_canShutdownLock` to make it very clear what is supposed to be locked by the lock.